test/extended/util/framework.go

```diff
@@ -1447,6 +1447,13 @@ func GetRouterPodTemplate(oc *CLI) (*corev1.PodTemplateSpec, string, error) {
 		if !errors.IsNotFound(err) {
 			return nil, "", err
 		}
+		ds, err := k8sappsclient.DaemonSets(ns).Get("router-default", metav1.GetOptions{})
+		if err == nil {
```

Comment:
can error be nil and still have an empty DaemonSet?

Reply:
No error means a valid `DaemonSet`.
Did you mean to ask for the case the `DaemonSet` doesn't exist and it would return nil here? 
In that case, there'd be a daemonset not found error returned.